api/src/main/java/net/starype/quiz/api/game/MaxGuessCounter.java

```diff
@@ -0,0 +1,30 @@
+package net.starype.quiz.api.game;
+
+import net.starype.quiz.api.game.player.UUIDHolder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MaxGuessCounter implements AnswerEligibility {
+
+    private int maxGuesses;
+    private Map<UUIDHolder, Integer> guessesPerPlayer;
+
+    public MaxGuessCounter(int maxGuesses) {
+        this.maxGuesses = maxGuesses;
+        this.guessesPerPlayer = new HashMap<>();
+    }
+
+    public void wrongGuess(UUIDHolder holder) {
+        guessesPerPlayer.put(holder, guessesPerPlayer.getOrDefault(holder, 0) + 1);
+    }
+
+    public void giveUp(UUIDHolder holder) {
+        guessesPerPlayer.put(holder, maxGuesses);
+    }
+
+    @Override
+    public boolean isEligible(UUIDHolder player) {
+        return guessesPerPlayer.getOrDefault(player, 0) < maxGuesses;

```

Comment:
Adding a condition if the ```maxGuesses``` is Integer.MAX_VALUE then always return true (use or)

Reply:
Yes I thought about maybe using `-1` as the value for unlimited tries

Reply:
I think using the MAX_VALUE is a better approaches as it is really a big value but your idea isn't bad. If so then add an
```java
return guessesPerPlayer.getOrDefault(player, 0) < maxGuesses || maxGuesses == -1;
return guessesPerPlayer.getOrDefault(player, 0) < maxGuesses || maxGuesses == Integer.MAX_VALUE;
```